src/CalculatorUI/App.xaml.cs

```diff
@@ -0,0 +1,516 @@
+        {
+            if (args.Kind == ActivationKind.Protocol)
+            {
+                // We currently don't pass the uri as an argument,
+                // and handle any protocol launch as a normal app launch.
+                OnAppLaunch(args, null);
+            }
+        }
+
+        internal void RemoveWindow(WindowFrameService frameService)
+        {
+            // Shell does not allow killing the main window.
+            if (m_mainViewId != frameService.GetViewId())
+            {
+                _ = HandleViewReleaseAndRemoveWindowFromMap(frameService);
+            }
+        }
+
+        internal void RemoveSecondaryWindow(WindowFrameService frameService)
+        {
+            // Shell does not allow killing the main window.
+            if (m_mainViewId != frameService.GetViewId())
+            {
+                RemoveWindowFromMap(frameService.GetViewId());
+            }
+        }
+
+        private static Frame CreateFrame()
+        {
+            var frame = new Frame();
+
+            // CSHARP_MIGRATION: TODO:
+            //frame.FlowDirection = LocalizationService.GetInstance().GetFlowDirection();
+
+            return frame;
+        }
+
+        private static void SetMinWindowSizeAndActivate(Frame rootFrame, Size minWindowSize)
+        {
+            // SetPreferredMinSize should always be called before Window.Activate
+            ApplicationView appView = ApplicationView.GetForCurrentView();
+            appView.SetPreferredMinSize(minWindowSize);
+
+            // Place the frame in the current Window
+            Window.Current.Content = rootFrame;
+            Window.Current.Activate();
+        }
+
+        private void OnAppLaunch(IActivatedEventArgs args, String argument)
+        {
+            // CSHARP_MIGRATION: TODO:
+            // Uncomment the following lines to display frame-rate and per-frame CPU usage info.
+            //#if _DEBUG
+            //    if (IsDebuggerPresent())
+            //    {
+            //        DebugSettings->EnableFrameRateCounter = true;
+            //    }
+            //#endif
+
+            args.SplashScreen.Dismissed += DismissedEventHandler;
+
+            var rootFrame = (Window.Current.Content as Frame);
+            WeakReference weak = new WeakReference(this);
+
+            float minWindowWidth = (float)((double)Resources[ApplicationResourceKeys.Globals.AppMinWindowWidth]);
+            float minWindowHeight = (float)((double)Resources[ApplicationResourceKeys.Globals.AppMinWindowHeight]);
+            Size minWindowSize = SizeHelper.FromDimensions(minWindowWidth, minWindowHeight);
+
+            ApplicationView appView = ApplicationView.GetForCurrentView();
+            ApplicationDataContainer localSettings = ApplicationData.Current.LocalSettings;
+            // For very first launch, set the size of the calc as size of the default standard mode
+            if (!localSettings.Values.ContainsKey("VeryFirstLaunch"))
```

Comment:
Regarding [this article](https://docs.microsoft.com/en-us/uwp/api/windows.foundation.collections.ipropertyset?view=winrt-19041)

`ApplicationDataContainer` is a `IPropertySet` Interface which derived from `IDictionary`

```
[Windows.Foundation.Metadata.ContractVersion(typeof(Windows.Foundation.FoundationContract), 65536)]
[Windows.Foundation.Metadata.Guid(2319707551, 62694, 17441, 172, 249, 29, 171, 41, 134, 130, 12)]
[Windows.Foundation.Metadata.HasVariant]
public interface IPropertySet : IDictionary<string,object>, IEnumerable<KeyValuePair<string,object>>, IObservableMap<string,object>
```

>This code snippet comes from the ControlChannelTrigger HTTP client sample. Note how the C# version casts to IDictionary<string,object> so that it can use the indexer, whereas the Visual C++ component extensions (C++/CX) version uses HasKey and Lookup.